### PR TITLE
Add curl --max-time

### PR DESCRIPTION
**Pull request by joshuar**
This PR and sets a default maximum execution time for a call to `curl` in the *nix script to 30s. This helps when, for example a call to a particular REST API end-point (like `/_segments`) takes a long time.  Ideally, getting most diagnostics back minus these calls which timeout is better than the end-user forcibly killing the script or Ctrl-C'ing when a command is taking too long.

### CODE REVIEW

File: bin/support-diagnostics.sh [hunk 1]
```diff
@@ -49,6 +51,7 @@ EOM
 	-a)  authType=$2;;
 	-c)  authCreds=$2;;
 	-p)  password=$2;;
+        -t)  curlTimeout=$2;;
```
Comment (pickypg) [inline bin/support-diagnostics.sh]: Extra leading space (not critical... but OCD).

**Comment by pickypg** [comment]
LGTM

**Comment by joshuar** [comment]
Will be deprecated with the upcoming 2.0 release.

**Comment by pickypg** [comment]
Did you intentionally close this PR rather than merge it in?

**Comment by joshuar** [comment]
Closed, because I thought we agreed to wait for the 2.0 (Java) branch, which incorporates these fixes anyway?

**Pull request closed without merging**